array_api_compat/common/_helpers.py

```diff
@@ -781,42 +781,28 @@ def _cupy_to_device(
     /,
     stream: int | Any | None = None,
 ) -> _CupyArray:
-    import cupy as cp  # pyright: ignore[reportMissingTypeStubs]
-    from cupy.cuda import Device as _Device  # pyright: ignore
-    from cupy.cuda import stream as stream_module  # pyright: ignore
-    from cupy_backends.cuda.api import runtime  # pyright: ignore
+    import cupy as cp
 
-    if device == x.device:
-        return x
-    elif device == "cpu":
+    if device == "cpu":
         # allowing us to use `to_device(x, "cpu")`
         # is useful for portable test swapping between
         # host and device backends
         return x.get()
```

Comment:
Whoa. I'm confused: this returns a numpy array if `x` was a cupy array. This might be intended, but then the return annotation is misleading? Same for the device argument annotation. So maybe add a comment if a numpydoc docstring with `Parameters` and `Returns` sections is an overkill.

Reply:
None of this has changed. See also #87.

Reply:
My point is that gh-87 is still open, and there are no tests, so if we touch this, great, let's take the opportunity to also improve testing.